packages/aws-cdk-lib/aws-events-targets/lib/ecs-task.ts

```diff
@@ -266,13 +266,20 @@ export class EcsTask implements events.IRuleTarget {
   }
 
   private createEventRolePolicyStatements(): iam.PolicyStatement[] {
-    const policyStatements = [new iam.PolicyStatement({
-      actions: ['ecs:RunTask'],
-      resources: [this.taskDefinition.taskDefinitionArn],
-      conditions: {
-        ArnEquals: { 'ecs:cluster': this.cluster.clusterArn },
-      },
-    })];
+    const policyStatements = [
+      new iam.PolicyStatement({
+        actions: ['ecs:RunTask'],
+        resources: [this.taskDefinition.taskDefinitionArn],
+        conditions: {
+          ArnEquals: { 'ecs:cluster': this.cluster.clusterArn },
+        },
+      }),
+      new iam.PolicyStatement({
+        actions: ['ecs:TagResource'],
+        // eslint-disable-next-line @aws-cdk/no-literal-partition
+        resources: [`arn:aws:ecs:*:*:task/${this.cluster.clusterName}/*`],
```

Comment:
open to ideas on this. I don't think there is a good way to get region? hence the `*` for region in arn.

Reply:
I actually think no Region may be the way to go here.. since it's a role/policy.

Reply:
`region` seems to be accessible from `this.cluster.env.region`

![image](https://github.com/aws/aws-cdk/assets/47882901/b74302ac-9ec9-435c-8f30-acaacb6c5d7d)